ouroboros-network/src/Ouroboros/Network/StrictFingerTree.hs

```diff
@@ -0,0 +1,346 @@
+{-# LANGUAGE BangPatterns               #-}
+{-# LANGUAGE CPP                        #-}
+#if __GLASGOW_HASKELL__ >= 706
```

Comment:
We don't care about compatibility with GHC (and base) versions other than the current one (8.6, in the future 8.8, I expect), so this can go.